TSRM/TSRM.c

```diff
@@ -222,14 +232,46 @@ TSRM_API void tsrm_shutdown(void)
 	tsrm_new_thread_begin_handler = NULL;
 	tsrm_new_thread_end_handler = NULL;
 	tsrm_shutdown_handler = NULL;
+
+	tsrm_reserved_pos  = 0;
+	tsrm_reserved_size = 0;
 }/*}}}*/
 
 
-/* allocates a new thread-safe-resource id */
-TSRM_API ts_rsrc_id ts_allocate_id(ts_rsrc_id *rsrc_id, size_t size, ts_allocate_ctor ctor, ts_allocate_dtor dtor)
+/* enlarge the arrays for the already active threads */
+static void tsrm_update_acrive_threads(void)
```

Comment:
s/acrive/active

Reply:
Thanks for review and for GC/MM idea. I'll check it.

Reply:
@nikic 
Just ZEND_TLS usage is not enough.
TSRM also calls ctor/dtor, when necessary.